jcl/src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -529,8 +529,59 @@ public StringBuffer append (float value) {
  * @param		value	the integer
  * @return		this StringBuffer
  */
-public StringBuffer append (int value) {
-	return append(Integer.toString(value));
+public synchronized StringBuffer append(int value) {
+	/*[IF Sidecar19-SE]*/
+	/*[IF Sidecar19-SE-B174]*/
+	int currentLength = lengthInternalUnsynchronized();
+	int currentCapacity = capacityInternal();
+			
+	int valueLength = Integer.stringSize(value);
+			
+	int newLength = currentLength + valueLength;
+			
+	if (newLength > currentCapacity) {
+		ensureCapacityImpl(newLength);
+	}
+
+	if (String.enableCompression && count >= 0) {
+		Integer.getChars(value, newLength, this.value);
+	} else {
+		StringUTF16.getChars(value, newLength, this.value);
+	}
+
+	count = newLength;
+			
+	return this;
+	/*[ELSE]*/
+	return append (String.valueOf (value));
+	/*[ENDIF]*/
+	/*[ELSE]*/
+	if (value != Integer.MIN_VALUE) {
+		if (String.enableCompression && count >= 0) {
+			return append (String.valueOf (value));
```

Comment:
Just for completeness. This can also be fastpath, and likely should be but Java 8 does not currently have an out-of-the-box API for String compression where we can pass a `byte[]` which gets filled with the string representation of an integer. This is effectively what `StringUTF16.getChars(II[B)` does a few lines above.

To get this working we can implement our own version of such an API in an internal J9 JCL helper class and call it since it is a static method. It would however have to be a clean room implementation.